src/Symfony/Component/DependencyInjection/Tests/EnvVarProcessorTest.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace Symfony\Component\DependencyInjection\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\EnvVarProcessor;
+use Symfony\Component\DependencyInjection\Exception\LogicException;
+
+class EnvVarProcessorTest extends TestCase
+{
+    const TEST_CONST = 'test';
+
+    public function testGetEnvString()
+    {
+        $container = new ContainerBuilder();
+        $container->setParameter('env(foo)', 'hello');
+        $container->compile();
+
+        $processor = new EnvVarProcessor($container);
+
+        $result = $processor->getEnv('string', 'foo', function () {
+            throw new LogicException('Shouldnt be called');
+        });
+
+        $this->assertSame('hello', $result);
+    }
+
+    public function testGetEnvBool()
+    {
+        $processor = new EnvVarProcessor(new Container());
+
+        $result = $processor->getEnv('bool', 'foo', function ($name) {
+            $this->assertSame('foo', $name);
+
+            return '1';
+        });
+
+        $this->assertTrue($result);
```

Comment:
Just asking: `assertTrue()` accepts as true anything that PHP considers true ... or does it a strict `true === ...` comparison? We're getting a bool value here, so we need to be sure that a boolean is returned.

Reply:
@javiereguiluz It's a strict comparison.

https://github.com/sebastianbergmann/phpunit/blob/ea739aae4742aa4eae1e98d4d06acdbf01692415/src/Framework/Constraint/IsTrue.php#L25-L28